packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -31,3 +32,10 @@ class _WindowsUtils implements OperatingSystemUtils {
     return new ProcessResult(0, 0, null, null);
   }
 }
+
+Future<int> findAvailablePort() async {
+  ServerSocket socket = await ServerSocket.bind(InternetAddress.LOOPBACK_IP_V4, 0);
+  int port = socket.port;
+  await socket.close();
+  return port;
```

Comment:
Yeah, we're not the ones binding to the port, but are instead looking for an available port to bind to.  Don't know of a better way...